cmake/engine_git.cmake

```diff
@@ -1,50 +1,83 @@
-function (set_git_version BRANCH HASH VERSION)
+function (set_git_version BRANCH HASH VERSION_UNFORMATTED)
     set(OBENGINE_GIT_BRANCH ${BRANCH} CACHE STRING "ObEngine git branch" FORCE)
     set(OBENGINE_GIT_COMMIT_HASH ${HASH} CACHE STRING "ObEngine git commit hash" FORCE)
-    set(OBENGINE_VERSION ${VERSION} CACHE STRING "ObEngine version" FORCE)
+
+    string(REGEX MATCH "v(.+)\\.(.+)\\.(.+)$" VERSION "${VERSION_UNFORMATTED}")
+
+    set(OBENGINE_VERSION ${VERSION} CACHE STRING "ObEngine version string" FORCE)
+    set(VERSION_MAJOR ${CMAKE_MATCH_1})
+    set(VERSION_MINOR ${CMAKE_MATCH_2})
+    set(VERSION_PATCH ${CMAKE_MATCH_3})
+
+    string(REGEX MATCH "([0-9]+)$" RES ${VERSION_MAJOR})
+    if(NOT ${RES} STREQUAL "")
+        set(VERSION_MAJOR ${CMAKE_MATCH_1})
+        # 0*([0-9]+)$ seems nicer, but CMake can not lazily match
+        math(EXPR VERSION_MAJOR "${VERSION_MAJOR} * 1")
+    else()
+        set(VERSION_MAJOR "-1")
```

Comment:
Could we have "?" instead of "-1" ? Seems more clear to me

Reply:
Versions are currently stored as 3 integers in C++, we can of course change this to a "string"-based storage.

Reply:
Or std::optional<int> but that seems overkill, let's stay with the -1 :)